compiler/native/script.go

```diff
@@ -98,16 +98,21 @@ func generateScriptPosix(commands []string) string {
 // setupScript is a helper script this is added to the build to ensure
 // a minimum set of environment variables are set correctly.
 const setupScript = `
-cat <<EOF > $HOME/.netrc
-machine $VELA_NETRC_MACHINE
-login $VELA_NETRC_USERNAME
-password $VELA_NETRC_PASSWORD
+set +e
+touch $HOME/.netrc 2>/dev/null
+if [ -w "$HOME" ]; then
+  cat <<EOF > $HOME/.netrc
+machine ${VELA_NETRC_MACHINE:-}
+login ${VELA_NETRC_USERNAME:-}
+password ${VELA_NETRC_PASSWORD:-}
```

Comment:
why the `:-`?

Reply:
Meh, just a substitution check that I don't think is necessary given we always set these. I'll remove it

Reply:
even if it's not set it should result in empty value either way afaik, but sounds good